src/main/java/bio/terra/app/controller/UnauthenticatedApiController.java

```diff
@@ -68,7 +68,10 @@ public UnauthenticatedApiController(
     @Override
     public ResponseEntity<Void> shutdownRequest() {
         try {
-            jobService.shutdown();
+            if (!jobService.shutdown()) {
+                // Shutdown did not complete. Return an error so the caller knows that
```

Comment:
I understand that shutdown() returning false means that stairway didn't quietdown or terminate successfully in the time allotted. But what is the state of the world when that happens? Does it mean that the K8S pod shutting down and stairway terminating are now in a race condition and it's possible we'll need to cleanup flights in a bad state with some cleanup job?

Reply:
Good question. There are two kinds of shutdown:
1. The tidy kind where we are able to close out all flights and queue them up for other (later) Stairway instances to run.
2. The messy kind where flights are left in the RUNNING state and need to be recovered.
The boolean is reporting whether things are tidy or messy.

DR-835 is the ticket I'm working on now that implements the recovery from the messy state. We have always had that recovery in Stairway, but the flow of it changes when we have multiple Stairway instances running and failing.